### PR TITLE
[#11878] Update Admin Home Page UI for ARF

### PR DESCRIPTION
**Pull request by domoberzin**
Part of #11878 

**Outline of Solution**
This PR contains the updates for the Admin Home Page UI for the Account Request Form changes.

This PR creates a account request table component to display account requests. As of this current PR, the only features that come with the ability to delete the account request or view the comment of the account request. Further features (approve, edit, reject) will be added in a later PR. Since this table will be re-used for the admin search page later on, the searchString is used to distinguish between the table layout for both pages.

There is a minimum height and width specified on the comments box to handle cases of extremely long comments.

As per the discussion below, the screenshots and recordings below are slightly outdated, the current version will not have a separate country column, i.e. institute and country will be combined into one column, named `Institute, Country`. These screenshots/clips will be updated later on.

Screenshot of a table with no long comments (not scrollable)
Screenshot 2024-03-25 at 12 47 42 PM

Screenshot of the view comment modal
Screenshot 2024-03-25 at 1 11 08 PM

Recording of the table being horizontally scrollable (only when there exists a long comment)
<video src="https://github.com/TEAMMATES/teammates/assets/74132255/5707ba1c-753d-4742-a06d-8a7ebafd1153">

Recording of the delete feature of the table
<video src="https://github.com/TEAMMATES/teammates/assets/74132255/9ada8599-a454-461d-920b-25f4334921ed">

### CODE REVIEW

File: src/web/app/components/account-requests-table/account-requests-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  email: string;
+  status: string;
+  institute: string;
+  country: string;
+  createdAtText: string;
+  registeredAtText: string;
+  comments: string;
+  registrationLink: string;
+  showLinks: boolean;
+}
```
Comment (jayasting98) [inline src/web/app/components/account-requests-table/account-requests-table.component.ts:25]: Maybe we can move this out into a separate file so that this one is only concerned with the component. Also, I think we usually call this `XyzRowModel`. You can take a look at the [logs table](https://github.com/TEAMMATES/teammates/blob/11b8b818f3a1e2b810de20907725561b009f9297/src/web/app/components/logs-table/logs-table-model.ts) for an example.

File: src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts [hunk 2]
```diff
@@ -236,6 +247,45 @@ export class AdminHomePageComponent {
     );
   }
 
+  private formatTimestampAsString(timestamp: number, timezone: string): string {
+      const dateFormatWithZoneInfo: string = 'ddd, DD MMM YYYY, hh:mm A Z';
+
+      return this.timezoneService
+          .formatToString(timestamp, timezone, dateFormatWithZoneInfo);
+  }
```
Comment (jayasting98) [inline src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts:255]: Is it possible to use the [`FormatDetailDatePipe`](https://github.com/TEAMMATES/teammates/blob/11b8b818f3a1e2b810de20907725561b009f9297/src/web/app/components/teammates-common/format-date-detail.pipe.ts) instead of this?

File: src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts [hunk 3]
```diff
@@ -236,6 +247,45 @@ export class AdminHomePageComponent {
...
+
+  private formatAccountRequests(requests: AccountRequests): AccountRequestData[] {
+    const timezone: string = this.timezoneService.guessTimezone() || 'UTC';
+    return requests.accountRequests.map((request) => {
+      const [institute, country] = request.institute.split(', ').length === 2
+      ? request.institute.split(', ') : [request.institute, ''];
```
Comment (jayasting98) [inline src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts:261]: I'm not sure how well this would work. Let's say `request.institute` is `` `${instituteAbc}, ${countryXyz}` ``. Suppose `instituteAbc` had commas in it, then the length after doing `split` would be more than two, so the value assigned to `[institute, country]` would be ``[`${instituteAbc}, ${countryXyz}`, '']``, so `countryXyz` would not be in the correct column. This should be similar for the case where `countryXyz` had commas in it.

File: src/web/services/account.service.ts [hunk 4]
```diff
@@ -12,7 +20,9 @@ import { AccountCreateRequest } from '../types/api-request';
   providedIn: 'root',
 })
 export class AccountService {
-  constructor(private httpRequestService: HttpRequestService) {
+  constructor(
+    private httpRequestService: HttpRequestService,
+    ) {
```
Comment (jayasting98) [inline src/web/services/account.service.ts:25]: I think this seems like an unnecessary change?

**Review by jayasting98 — CHANGES_REQUESTED** [review]
It seems alright overall, but I have a few comments.

File: src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts (hunk 3 above)
Comment (domoberzin) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: Thanks for the feedback, I'm also not sure whats an easy way to resolve this, but I changed it to split at the last comma instead of  the first, so this at least guarantees a length of 2, and also since it is more likely that the institute contains a comma, as compared to the country. Otherwise, do you have any suggestions on how to handle this?

Comment (jayasting98) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: For me, I feel like it's easier to just keep it as one institute column, instead of splitting it into two columns, but I'm not sure what's the best option either.

@ziqing26, @EuniceSim142, do you have any thoughts on this?

Comment (domoberzin) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: Here's the current code for reference 
```
const lastCommaIndex = request.institute.lastIndexOf(', ');
const [institute, country] = lastCommaIndex === -1
        ? [request.institute, '']
        : [request.institute.slice(0, lastCommaIndex), request.institute.slice(lastCommaIndex + 2)];
```

Comment (jayasting98) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: I feel like what you suggested where you split at the last comma may not be too bad actually. You're right that a comma in a country's name is rare.

It seems like there is currently only one country with a comma in their name on [this list](https://worldpopulationreview.com/country-rankings/official-names-of-countries), and it's in their official name, "Nation of Brunei, Abode of Peace", which I doubt anybody is likely to use for TEAMMATES. Besides, it seems as though it should be "Brunei Darussalam" instead? Also, after skimming [the list on Wikipedia](https://en.wikipedia.org/wiki/List_of_sovereign_states), I could not even find any countries with commas in their names.

So, yea, we can probably settle with what you suggested, unless the others have anything to say about it. Just hope that no new country appears with a comma in its name, or no existing country renames itself. 😁

With this, I wonder if we can just migrate towards having an actual country column in the database. I doubt there are many cases with multiple commas to begin with, so those can be corrected manually. But anyway, this is out of the scope of this PR.

Comment (ziqing26) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: Let's keep it to be displayed as one column in the table i.e. "Institute, Country"
Because (1) currently the db's institute field contain both institute and country (i.e. it is a string editable by admin) (2) we are not using dropdown in the frontend for this field due to sensitivity of region v.s. country in some scenarios, so no guarantee on the format of country string

Comment (domoberzin) [reply src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts]: Updated with the above-mentioned changes, so the institute and country variables are now combined as `instituteAndCountry` on the `AccountRequestTableRowModel`

**Comment by jayasting98** [comment]
Don't forget to update the screenshots and clips in the original comment, for anybody else who comes to review.

**Review by jayasting98 — APPROVED** [review]
Good work. Thanks!

**Review by ziqing26 — APPROVED** [review]
Thank you for the amazing work! 💪  LGTM
